velox/exec/PrefixSort.cpp

```diff
@@ -185,7 +216,8 @@ int PrefixSort::comparePartNormalizedKeys(char* left, char* right) {
   // If prefixes are equal, compare the remaining sort keys with rowContainer.
   char* leftRow = getRowAddrFromPrefixBuffer(left);
   char* rightRow = getRowAddrFromPrefixBuffer(right);
-  for (auto i = sortLayout_.numNormalizedKeys; i < sortLayout_.numKeys; ++i) {
+  for (auto i = sortLayout_.comparisonStartIndex; i < sortLayout_.numKeys;
```

Comment:
Wonder why PrefixSortTest cannot cover this change, it can pass all the tests when the start index is sortLayout_.numNormalizedKeys. Can you please add a test to cover this?

Reply:
Avoiding future change may break this logic.

Reply:
The new added cases in `singleKey` and `singleKeyWithNulls` should catch this issue, but they didn't. I think this is caused by a bug in PrefixSortTest: Only 'double', 'timestamp' and 'string' are executed in the loop. I fixed it to cover all cases in this PR.
https://github.com/facebookincubator/velox/blob/86d6f3389c12759efd6c4fa212004156b5f0f7df/velox/exec/tests/PrefixSortTest.cpp#L212C2-L212C41

Reply:
Do you try if the test fail without this change?

Reply:
Yes.

Reply:
Ok, thanks for your clarification.